TST: Check map function works with StringDtype (#40823)

mdhsieh opened the pull request:
Checks map function output on 2 DataFrames with data type StringDtype.

- [ ] closes #40823
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

mdhsieh committed:
TST: Check map function works with StringDType (#40823).

mdhsieh committed:
TST: Remove unused imports.

jreback commented on pandas/tests/frame/methods/test_map.py:
```diff
@@ -0,0 +1,24 @@
+import pandas._testing as tm
+
+
+class TestMap:
```
we have extensive map tests here: pandas/tests/apply/test_series_apply.py

see if you can find something similar. you can use the fixture `any_string_dtype` to parmeterize this over various kinds of strings.

mdhsieh replied:
Ok moved. Should the fixture be used like this? 
```
# parameter
test_series_map_stringdtype(any_string_dtype):
    ser1 = Series(
        # ...
        # then set here for all Series
        dtype=any_string_dtype,
    )
 ```

jreback commented on pandas/tests/frame/methods/test_map.py:
```diff
@@ -0,0 +1,24 @@
+            dtype=pd.StringDtype(),
+        )
+
+        tm.assert_frame_equal(result, expected)
```
this is a series method, so test on that.

mdhsieh committed:
TST: Remove new test file since want to add test in existing file.

mdhsieh committed:
TST: Add new test to pandas/tests/apply/test_series_apply.py

mdhsieh committed:
TST: Add updated test on Series (#40823).

Checking map function on two series of data type StringDType.
Series data is arbitrary.

mdhsieh committed:
TST: Change StringDType to any string data type.

Using suggested test fixture any_string_dtype.

jreback commented:
thanks @mdhsieh
